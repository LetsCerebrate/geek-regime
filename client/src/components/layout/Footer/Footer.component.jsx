import React from "react";
import moment from "moment";

import styles from "./Footer.module.scss";

function Footer () {
    const currentYear = moment().format("YYYY");

    return (
        <footer className={styles.container}>
            <section className={styles.content}>
                All rites reversed. {currentYear}.
            </section>
        </footer>
    );
}

export default Footer;
