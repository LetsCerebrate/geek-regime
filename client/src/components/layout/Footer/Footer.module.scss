@import "styles/style.scss";

.container {
    grid-area: footer;
    z-index: 1;
    display: grid;
    grid-template-areas: ". content .";
    grid-template-rows: auto;
    grid-template-columns: 20% 1fr 20%;
    background-color: $footer-content-bg-color;

    .content {
        grid-area: content;
        display: flex;
        align-items: center;
        padding: 2rem 3vw;
        color: set-text-color($footer-content-bg-color);
    }
}
