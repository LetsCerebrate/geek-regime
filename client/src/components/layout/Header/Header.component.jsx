import React from "react";

import Logo from "components/layout/Logo";
import styles from "./Header.module.scss";

function Header () {
    return (
        <header className={styles.container}>
            <section className={styles.content}>
                <Logo />
            </section>
        </header>
    );
}

export default Header;
