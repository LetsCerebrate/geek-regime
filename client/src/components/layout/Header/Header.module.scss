@import "styles/style.scss";

.container {
    grid-area: header;
    z-index: 1;
    display: grid;
    grid-template-areas: ". content .";
    grid-template-rows: auto;
    grid-template-columns: 20% 1fr 20%;
    background-color: $header-bg-color;
    box-shadow:
        0 1px 0.8rem rgba(0, 0, 0, 0.2),
        0 0 1.5rem rgba(0, 0, 0, 0.04) inset;

    .content {
        grid-area: content;
        display: flex;
        padding: 0 3vw;
    }

    @media screen and (max-width: 1024px) {
        & {
            display: none;
        }
    }
}
