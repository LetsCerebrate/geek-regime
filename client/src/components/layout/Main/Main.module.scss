@import "styles/style.scss";

.container {
    grid-area: main;
    display: flex;
    box-sizing: border-box;
    justify-content: center;

    .content {
        box-sizing: border-box;
        width: 100%;
        padding: 2rem 1rem;
    }

    @media screen and (min-width: 1600px) {
        .content {
            width: 1024px;
        }
    }
}
