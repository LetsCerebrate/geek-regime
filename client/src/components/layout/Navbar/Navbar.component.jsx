import React from "react";

import Logo from "components/layout/Logo";
import NavbarList from "components/layout/NavbarList";
import styles from "./Navbar.module.scss";

function Navbar () {
    return (
        <div className={styles.container}>
            <div className={styles.mobileLogo}>
                <Logo isMobileView />
            </div>

            <div className={styles.desktopNavbar}>
                <NavbarList />
            </div>
        </div>
    );
}

export default Navbar;
