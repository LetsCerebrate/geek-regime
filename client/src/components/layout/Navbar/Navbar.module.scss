@import "styles/style.scss";

.container {
    grid-area: navbar;
    display: flex;
    justify-content: flex-end;

    .desktopNavbar {
        width: 17rem;
    }

    .mobileLogo {
        display: none;

        a {
            padding: 0;
            transition: color 50ms ease;
        }
    }

    @media screen and (max-width: 1024px) {
        justify-content: flex-start;

        .mobileLogo {
            display: flex;
            margin-left: 3vw;

            a:hover {
                color: $mobile-logo-text-color--hover !important;
            }
        }

        @media screen and (max-width: 500px) {
            .desktopNavbar {
                display: none;
            }

            .mobileLogo {
                margin-left: 0;
            }
        }
    }
}
