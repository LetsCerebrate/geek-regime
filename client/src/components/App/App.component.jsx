import { Route, Switch } from "react-router-dom";
import React, { Suspense, lazy, useEffect } from "react";
import { connect } from "react-redux";

import {
    DRAFT,
    POST,
    PROFILE,
    SIGN_IN
} from "utils/const/pathnames";

import { PrivateRoute } from "components/Route";
import ErrorBoundary from "components/ErrorBoundary";
import Footer from "components/Footer";
import Header from "components/Header";
import Main from "components/Main";
import Nav from "components/Nav";
import Spinner from "components/Spinner";
import { checkSessionStart } from "redux/session/session.actions";
import { propTypes } from "./App.props";
import styles from "./App.module.scss";

const DraftEditor = lazy(() => import("pages/DraftEditor"));
const Home = lazy(() => import("pages/Home"));
const Post = lazy(() => import("pages/Post"));
const Profile = lazy(() => import("pages/Profile"));
const SignIn = lazy(() => import("pages/SignIn"));

App.propTypes = propTypes;

function App ({ onCheckSessionStart }) {
    useEffect(() => {
        onCheckSessionStart();
    }, [onCheckSessionStart]);

    return (
        <div className={styles.container}>
            <Header />

            <Main>
                <ErrorBoundary>
                    <Suspense fallback={<Spinner />}>
                        <Switch>
                            <Route
                                component={Home}
                                exact
                                path="/"
                            />

                            <Route
                                component={SignIn}
                                path={SIGN_IN}
                            />

                            <PrivateRoute
                                component={Profile}
                                path={PROFILE}
                            />

                            <PrivateRoute
                                component={DraftEditor}
                                path={DRAFT}
                            />

                            <PrivateRoute
                                component={DraftEditor}
                                path={`${POST}/:id/edit`}
                            />

                            <Route
                                component={Post}
                                path={`${POST}/:id`}
                            />
                        </Switch>
                    </Suspense>
                </ErrorBoundary>
            </Main>

            <Nav />
            <Footer />
        </div>
    );
}

const mapDispatchToProps = (dispatch) => ({
    onCheckSessionStart: () => dispatch(checkSessionStart())
});

const ConnectedApp = connect(
    null,
    mapDispatchToProps
)(App);

export default ConnectedApp;
