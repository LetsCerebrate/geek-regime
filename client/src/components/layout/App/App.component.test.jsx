import React from "react";
import { shallow } from "enzyme";
import toJson from "enzyme-to-json";

import { App } from "./App.component";
import Footer from "components/layout/Footer";
import Header from "components/layout/Header";
import Main from "components/layout/Main";
import Navbar from "components/layout/Navbar";

let wrapper = null;

beforeEach(() => {
    const mockProps = {
        onCheckSessionStart: jest.fn(),
        onFetchPostsStart: jest.fn()
    };

    wrapper = shallow(
        <App {...mockProps} />
    );
});

it("should render App", () => {
    expect(toJson(wrapper)).toMatchSnapshot();
});

it("should render child components", () => {
    const header = wrapper.find(Header);
    const nav = wrapper.find(Navbar);
    const main = wrapper.find(Main);
    const footer = wrapper.find(Footer);

    expect(header.exists()).toBe(true);
    expect(nav.exists()).toBe(true);
    expect(main.exists()).toBe(true);
    expect(footer.exists()).toBe(true);
});
