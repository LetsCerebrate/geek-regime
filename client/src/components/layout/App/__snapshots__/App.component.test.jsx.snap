// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render App 1`] = `
<div
  className="container"
>
  <Header />
  <Main>
    <ErrorBoundary>
      <Suspense
        fallback={<Spinner />}
      >
        <Switch>
          <Route
            component={
              Object {
                "$$typeof": Symbol(react.lazy),
                "_ctor": [Function],
                "_result": null,
                "_status": -1,
              }
            }
            exact={true}
            path="/"
          />
          <Route
            component={
              Object {
                "$$typeof": Symbol(react.lazy),
                "_ctor": [Function],
                "_result": null,
                "_status": -1,
              }
            }
            path="/sign-in"
          />
          <Connect(PrivateRoute)
            component={
              Object {
                "$$typeof": Symbol(react.lazy),
                "_ctor": [Function],
                "_result": null,
                "_status": -1,
              }
            }
            path="/profile"
          />
          <Connect(PrivateRoute)
            component={
              Object {
                "$$typeof": Symbol(react.lazy),
                "_ctor": [Function],
                "_result": null,
                "_status": -1,
              }
            }
            path="/draft"
          />
          <Connect(PrivateRoute)
            component={
              Object {
                "$$typeof": Symbol(react.lazy),
                "_ctor": [Function],
                "_result": null,
                "_status": -1,
              }
            }
            path="/post/:id/edit"
          />
          <Route
            component={
              Object {
                "$$typeof": Symbol(react.lazy),
                "_ctor": [Function],
                "_result": null,
                "_status": -1,
              }
            }
            path="/post/:id"
          />
          <Route
            component={
              Object {
                "$$typeof": Symbol(react.lazy),
                "_ctor": [Function],
                "_result": null,
                "_status": -1,
              }
            }
          />
        </Switch>
      </Suspense>
    </ErrorBoundary>
  </Main>
  <Nav />
  <Footer />
</div>
`;
